File: domain/src/main/java/com/github/mrmitew/bodylog/domain/repository/Repository.java
package com.github.mrmitew.bodylog.domain.repository;

import com.github.mrmitew.bodylog.domain.repository.entity.Profile;

import io.reactivex.Completable;
import io.reactivex.Observable;

public interface Repository {
    Observable<Profile> getProfile();

    Observable<Profile> getProfileRefreshing();

    Completable setProfile(Profile profile);
}

File: domain/src/main/java/com/github/mrmitew/bodylog/domain/repository/Repository.kt
package com.github.mrmitew.bodylog.domain.repository

import com.github.mrmitew.bodylog.domain.repository.entity.Profile

import io.reactivex.Completable
import io.reactivex.Observable

interface Repository {
    fun getProfile(): Observable<Profile>

    fun getProfileRefreshing(): Observable<Profile>

    fun setProfile(profile: Profile): Completable
}
